### DISCUSSION TITLE
More precise NumberInput when dealing with long decimals or big numbers

### DISCUSSION THREAD

**Discussion by songkeys** [mantinedev/mantine#5421] [Feature requests]
Currently, the `NumberInput` uses JavaScript numbers to process the input, but this can cause problems for long decimals or big integers (e.g., when the input is greater than `Number.MAX_SAFE_INTEGER`).

Most people may not encounter this issue; but if so, it really matters. Perhaps a documented attention is needed?

To solve this, we might consider implementing with a math library or using `bigint` along with manual parsing. What are your thoughts?

**Comment by rtivital** [comment]
For large numbers use TextInput instead and perform conversion on your side. I do not really want to maintain this feature in NumberInput.

**Reply by songkeys** [reply]
Using TextInput could do, but developers have to implement the "separators", "decimals", "controls", "min/max"... features in NumberInput again. But it's doable anyway.

I understand the big numbers case is rare in most applications. I'll just throw this discussion here and see if there is any future needs by others.
